### ISSUE TITLE
Allow both Published and Accepted version as `preferred_version`

### ISSUE THREAD

**Issue by ajkiessl** [psu-libraries/researcher-metadata#789]
This will require adding a third potential `preferred_version`.  We can call it "Published or Accepted".  The permissions data import will have to be modified so that if a publication can have both, this value is set.  We'll also have to alter the logic for what constitutes a "failed" version check/match.

**Comment by anaelizabethenriquez** [comment]
Thinking about this more, I think I got confused before. Sorry about that. There may be different values for License, Set Statement, and Embargo depending on whether it is the Published Version or the Accepted Version being deposited. So are we sending everything like this for manual permissions review once we figure out which version the file from Activity Insight is, and expecting an admin to update the preferred version to whatever we've got and fill in permissions metadata as needed? @ajkiessl

**Comment by EricDurante** [comment]
@anaelizabethenriquez Hmm, well the way that we have things working now, license, set statement, and embargo date are properties of the publication in RMD rather than properties of each of the files, and they would be the values for whatever the "preferred version" (also a property of the publication) is.

In the case where either version can be deposited, I guess that there are a few ways that it could be handled. I think that the least nice way would be to leave things as they are and do what you described - expect the admins to manually check and possibly correct the permissions metadata for whichever version is actually being deposited.

A nicer way might be to tweak our data model and do some additional metadata importing. One way to do this might be to make license, set statement, and embargo date properties of the files and then also change the workflow to automatically collect this metadata for each file _after_ we know the file version. However, I think that we could also do something similar without changing the existing workflow. We could model permissions metadata in the same way that OA Button does. In other words, a publication in RMD would could have multiple sets of permissions (a "permission" would be its own data model with the license, etc. properties), and one or more of those could be marked as "preferred" or "acceptable". We could gather all of this metadata from OA Button during the existing automated permissions check step and then use it during the final steps to determine if we have an acceptable version. If we do, then we'd already have the right metadata for that version, and we could automatically use that set of permissions for the chosen file. I don't have a strong feeling about which of these two approaches would be better yet..

**Comment by ajkiessl** [comment]
Yeah, the second option sounds better, but would likely be more time consuming.  I think it could get done in the next two sprints, though.  I don't think the first option would require any _big_ changes, and could hold us off until we get to another round of development.

I'm inclined to do the second option just so things are more thoroughly done from the start, but if there are other high priority things we want to get done, and the first option suffices in the meantime, then it might be better to do option 1.

**Comment by EricDurante** [comment]
I think I do too. They'd probably be kind of similar in terms of difficulty of implementation, but the second option seems like it might be a little easier. It likely has some implications for the workflow that I'd have to think through. For instance, if either version of a publication could be deposited and we didn't yet know which file version we had, would we want to require all of the permissions metadata for both versions to be filled in before allowing the publication to move on? And if we already knew which file version we had, and we had complete permissions metadata for that version but not the other, would we proceed then? (just thinking about expressing this in plain English makes my head hurt a little bit, let alone in SQL 😅).

**Comment by EricDurante** [comment]
Thanks @ajkiessl, I was just about to post a similar note.

One initial thought about this approach - it's definitely going to upend the current process for doing manual permissions curation which is based around having the permissions metadata in the publication records. I don't think that'll be a huge deal, but it's something that we'll need to consider.

**Comment by anaelizabethenriquez** [comment]
Ah, yes, I see. It is nice for the admin updating the permissions metadata in the manual review process to be able to do that on the publication record, where they can clearly see the other metadata (title, journal, DOI, etc.) for the publication. If the permissions metadata is in another record, we'll need to be careful that we know the publication we are researching permissions metadata for. That said, we wouldn't need to edit those publication fields, so maybe there's an option where we could show the title, journal, and DOI from the linked publication record on the file record?

**Comment by EricDurante** [comment]
@anaelizabethenriquez, do you happen to know of an example of a publication in OA Button where both the accepted and published version can be deposited?

**Comment by EricDurante** [comment]
@anaelizabethenriquez - sorry, scratch that last question. It looks like we already have an example of such a publication in our fixture data.
